build-data-capturing-gradle-samples/capture-test-pts-support/gradle-test-pts-support.gradle

```
@@ -0,0 +1,77 @@
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+import java.util.Optional
+import java.util.jar.JarFile
+import java.util.stream.Stream
+import java.util.stream.Collectors
+import groovy.transform.Field
+
+/**
+ * This Gradle script captures Predictive Test Selection and Test Distribution compatibility for each Test task, 
+ * adding a flag as custom value.
+ * This should be applied to the root project since it configures all projects:
+ * <code> apply from: file('gradle-test-pts-support.gradle') </code>
+ */
+
+def buildScanApi = project.extensions.findByName('buildScan')
+if (!buildScanApi) {
+    return
+}
+
+@Field
+final def supportedEngines = [
+    'org.junit.support.testng.engine.TestNGTestEngine' : 'testng',
+    'org.junit.jupiter.engine.JupiterTestEngine' : 'junit-jupiter', 
+    'org.junit.vintage.engine.VintageTestEngine' : 'junit-vintage', 
+    'org.spockframework.runtime.SpockEngine' : 'spock',
+    'net.jqwik.engine.JqwikTestEngine' : 'jqwik',
+    'io.cucumber.junit.platform.engine.CucumberTestEngine' : 'cucumber',
+    'org.spekframework.spek2.junit.SpekTestEngine' : 'spek2'
+    ]
+
+allprojects {
+    tasks.withType(Test).configureEach { t ->
+        doFirst {
+            if (t.getTestFramework().getClass().getName() == 'org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestFramework') {
+                if (detectSupportedTestEngines(t).isEmpty()) {
+                    buildScanApi.value("${t.identityPath}", 'PTS_NO_SUPPORTED_ENGINE')
+                } else {
+                    buildScanApi.value("${t.identityPath}", 'PTS_SUPPORTED')
+                }
+            } else {
+                buildScanApi.value("${t.identityPath}", 'PTS_NO_JUNIT_PLATFORM')
+            }
+        }
+    }
+}
+
+Collection<String> detectSupportedTestEngines(Task t) {
+    try {
+        Stream<String> engines = t.classpath.files.stream()
+                .filter({ f -> f.name.endsWith('.jar')})
```

nit: you can omit parentheses here and below

```suggestion
                .filter { f -> f.name.endsWith('.jar')} 
```